src/python_framework_v02/troute/nhd_io.py

```diff
@@ -1085,3 +1085,38 @@ def build_coastal_ncdf_dataframe(coastal_ncdf):
     with xr.open_dataset(coastal_ncdf) as ds:
         coastal_ncdf_df = ds[["elev", "depth"]]
         return coastal_ncdf_df.to_dataframe()
+
+
+def lastobs_df_output(
+    lastobs_df,
+    dt,
+    nts,
+    t0,
+    gages,
+    lastobs_output_folder=False,
+):
+
+    # join gageIDs to lastobs_df
+    lastobs_df = lastobs_df.join(gages)
+
+    # timestamp of last simulation timestep
+    modelTimeAtOutput = t0 + timedelta(seconds = nts * dt)
+    modelTimeAtOutput_str = modelTimeAtOutput.strftime('%Y-%m-%d_%H:%M:%S')
+
+    # timestamp of last observation
+    var = [timedelta(seconds=d) for d in lastobs_df.time_since_lastobs.fillna(0)]
+    lastobs_timestamp = [modelTimeAtOutput - d for d in var]
+    lastobs_timestamp_str = [d.strftime('%Y-%m-%d_%H:%M:%S') for d in lastobs_timestamp]
+
+    # create xarray Dataset similarly structured to WRF-generated lastobs netcdf files
+    ds = xr.Dataset(
+        {
+            "stationId": (["stationIdInd"], lastobs_df["gages"].to_numpy(dtype = '|S15')),
+            "time": (["stationIdInd"], np.asarray(lastobs_timestamp_str,dtype = '|S19')),
+            "discharge": (["stationIdInd"], lastobs_df["lastobs_discharge"].to_numpy()),
+        }
+    )
+    ds.attrs["modelTimeAtOutput"] = "example attribute"
+
+    # write-out LastObs file as netcdf
+    ds.to_netcdf(lastobs_output_folder + "nudgingLastObs." + modelTimeAtOutput_str + ".nc")
```

Comment:
Need to use pathlib here to join this path -- ended up with the humorous: 
`outputnudgingLastObs.2018-09-01_01:00:00.nc`